src/mail/mail.service.ts

```diff
@@ -0,0 +1,130 @@
+import * as ejs from "ejs";
+import { Injectable, Inject } from "@nestjs/common";
+import { SendEmailParamsDto } from "./dto/send-email-params.dto";
+import { MailResponse } from "./dto/mail-response";
+import { MailCoreService } from "./mail-core.service";
+import { ResponseStatus } from "../common/dto/response-base";
+
+@Injectable()
+export class MailService {
+  private readonly emailFromString = `${process.env.MAILGUN_DEFAULT_NAME} <${process.env.MAILGUN_DEFAULT_EMAIL}>`;
+
+  constructor(
+    @Inject(MailCoreService) private mailCoreService: MailCoreService
+  ) {}
+
+  /**
+   * Sends an email with a text body to a single recipient.
+   * @param sendEmailParams SendEmailParamsDto
+   */
+  async sendTextEmail(
+    sendEmailParams: SendEmailParamsDto
+  ): Promise<MailResponse> {
+    if (!sendEmailParams.bodyText) {
+      throw new Error("Email params must include body text.");
+    }
```

Comment:
This function can also return a `Promise<void>`, The ResponseBase is mostly for Controllers sending back JSON to HTTP clients. Basically we don't need the try-catch, just return this.mailCoreService.sendMailPayload(payload);